ptf/platform_helper/brcm_t0_sai_helper.py

```diff
@@ -272,6 +289,485 @@ def turn_on_port_admin_state(self):
+            self.client,
+            route_entry=entry, 
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+
+        rif_id2 = sai_thrift_create_router_interface(self.client, 
+            src_mac_address=ROUTER_MAC,
+            type=SAI_ROUTER_INTERFACE_TYPE_PORT,
+            mtu=PC_PORT_MTU,
+            port_id=self.port2,
+            nat_zone_id=NAT_ZONE_ID)
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+
+        entry = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(PC_IP_32V4_PREFIX2),
+            switch_id=self.switch_id)
+
+        self.pc_32v4_route_entry2 = sai_thrift_create_route_entry(
+            self.client,
+            route_entry=entry, 
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+
+        entry = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(PC_IP_128V6_PREFIX2),
+            switch_id=self.switch_id)
+        self.pc_128v6_route_entry2 = sai_thrift_create_route_entry(
+            self.client,
+            route_entry=entry, 
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+
+
+        rif_id3 = sai_thrift_create_router_interface(self.client, 
+            src_mac_address=ROUTER_MAC,
+            type=SAI_ROUTER_INTERFACE_TYPE_PORT,
+            mtu=PC_PORT_MTU,
+            port_id=self.port3,
+            nat_zone_id=NAT_ZONE_ID)
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+
+        entry = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(PC_IP_32V4_PREFIX3),
+            switch_id=self.switch_id)
+
+        self.pc_32v4_route_entry3 = sai_thrift_create_route_entry(
+            self.client,
+            route_entry=entry, 
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+
+        entry = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(PC_IP_128V6_PREFIX3),
+            switch_id=self.switch_id)
+        self.pc_128v6_route_entry3 = sai_thrift_create_route_entry(
+            self.client,
+            route_entry=entry, 
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+
+
+        rif_id4 = sai_thrift_create_router_interface(self.client, 
+            src_mac_address=ROUTER_MAC,
+            type=SAI_ROUTER_INTERFACE_TYPE_PORT,
+            mtu=PC_PORT_MTU,
+            port_id=self.port4,
+            nat_zone_id=NAT_ZONE_ID)
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+
+        entry = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(PC_IP_32V4_PREFIX4),
+            switch_id=self.switch_id)
+
+        self.pc_32v4_route_entry4 = sai_thrift_create_route_entry(
+            self.client,
+            route_entry=entry, 
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+
+        entry = sai_thrift_route_entry_t(
```

Comment:
abstract those duplicate codes into the method and with necessary parameters, loop on the method with the condition.